File: izinga-ordermanager/src/main/java/io/curiousoft/izinga/ordermanagement/stores/event/StoreCreatedEvent.java
package io.curiousoft.izinga.ordermanagement.stores.event;

import io.curiousoft.izinga.commons.model.StoreProfile;
import lombok.Getter;
import org.springframework.context.ApplicationEvent;

@Getter
public abstract class StoreCreatedEvent extends ApplicationEvent {

    private final StoreProfile storeProfile;

    public StoreCreatedEvent(Object source, StoreProfile storeProfile) {
        super(source);
        this.storeProfile = storeProfile;
    }

}

File: izinga-ordermanager/src/main/java/io/curiousoft/izinga/ordermanagement/stores/event/StoreCreatedEvent.kt
package io.curiousoft.izinga.ordermanagement.stores.event

import io.curiousoft.izinga.commons.model.StoreProfile
import org.springframework.context.ApplicationEvent

abstract class StoreCreatedEvent(source: Any, val storeProfile: StoreProfile) :
    ApplicationEvent(
        source
    )